### ISSUE TITLE
Lint swapping/replacement using assignments.

### ISSUE THREAD

**Issue by Ticki** [Manishearth/rust-clippy#714]
This shoud be done using `std::mem::{swap, replace}` instead.

```rust
let mut a = 3;
let mut b = 4;
let tmp = a;
a = b;
b = tmp;
```

should generate a warning.
